src/main/java/in/twizmwaz/cardinal/command/TeleportCommands.java

```diff
@@ -69,8 +89,27 @@ public static void teleport(final CommandContext cmd, CommandSender sender) thro
                         if (cmd.getString(1).contains("~")) x += teleporting.getLocation().getX();
                         if (cmd.getString(2).contains("~")) y += teleporting.getLocation().getY();
                         if (cmd.getString(3).contains("~")) z += teleporting.getLocation().getZ();
-                        teleporting.teleport(new Location(teleporting.getWorld(), x, y, z, teleporting.getLocation().getYaw(), teleporting.getLocation().getPitch()));
-                        sender.sendMessage(ChatColor.YELLOW + "Teleported.");
+                        if ( -30000000 < x && x < 30000000) {
+                            if ( -30000000 < z && z < 30000000) {
+                                teleporting.teleport(new Location(teleporting.getWorld(), x, y, z, teleporting.getLocation().getYaw(), teleporting.getLocation().getPitch()));
+                                sender.sendMessage(ChatColor.YELLOW + new LocalizedChatMessage(ChatConstant.GENERIC_TELEPORTED).getMessage(((Player) sender).getLocale()));
+                                if ((Player) sender != teleporting) {
+                                teleporting.sendMessage(ChatColor.YELLOW + new LocalizedChatMessage(ChatConstant.GENERIC_TELEPORTED_BY, ((Player) sender).getDisplayName()).getMessage((teleporting.getLocale())));
+                                }
+                            } else {
+                                TranslatableComponent numTooBig = new TranslatableComponent("commands.generic.num.tooBig");
+                                numTooBig.addWith("" + z);
+                                numTooBig.addWith("30000000");
+                                numTooBig.setColor(net.md_5.bungee.api.ChatColor.RED);
+                                ((Player) sender).sendMessage(numTooBig);
+                            }
+                        } else {
+                            TranslatableComponent numTooBig = new TranslatableComponent("commands.generic.num.tooBig");
+                            numTooBig.addWith("" + x);
+                            numTooBig.addWith("30000000");
+                            numTooBig.setColor(net.md_5.bungee.api.ChatColor.RED);
+                            ((Player) sender).sendMessage(numTooBig);
```

Comment:
If this command uses `(Player) sender` so much we can probably just make it a local variable.